Purchases/Public/RCIntroEligibility.m

```diff
@@ -16,16 +16,18 @@ @interface RCIntroEligibility ()
 
 @implementation RCIntroEligibility
 
-- (instancetype)initWithEligibilityStatus:(RCIntroEligibilityStatus)status
-{
+- (instancetype)initWithEligibilityStatus:(RCIntroEligibilityStatus)status {
     if (self = [super init]) {
         self.status = status;
     }
     return self;
 }
 
-- (NSString *)description
-{
+- (instancetype)initWithEligibilityStatusCode:(int)statusCode {
+    return [self initWithEligibilityStatus:(RCIntroEligibilityStatus)statusCode];
+}
```

Comment:
this isn't strictly necessary, you could just do the casting on the callsite. But I figured it was a good idea to have it here so that we can add extra checks and assertions, and so that we can unit test it directly.